adapters/core2p2p/transaction.go

```diff
@@ -160,23 +157,42 @@ func adaptResourceLimits(bounds core.ResourceBounds) *spec.ResourceLimits {
 	}
 }
 
+func adaptResourceBounds(rb map[core.Resource]core.ResourceBounds) *spec.ResourceBounds {
+	return &spec.ResourceBounds{
+		L1Gas: adaptResourceLimits(rb[core.ResourceL1Gas]),
+		L2Gas: adaptResourceLimits(rb[core.ResourceL2Gas]),
+	}
+}
+
 func adaptDeployTransaction(tx *core.DeployTransaction) *spec.Transaction_Deploy_ {
 	return &spec.Transaction_Deploy_{
 		Deploy: &spec.Transaction_Deploy{
 			ClassHash:   AdaptHash(tx.ClassHash),
 			AddressSalt: AdaptFelt(tx.ContractAddressSalt),
 			Calldata:    AdaptFeltSlice(tx.ConstructorCallData),
+			Version:     0, // todo(kirill) remove field from spec? tx is deprecated so no future versions
```

Comment:
seems like it's decided to keep it because version is part of hashing for this tx type:
```
deploy_tx_hash = h(
    "deploy",
    version,
    contract_address,
    sn_keccak("constructor"),
    h constructor_calldata),
    0,
    chain_id
)
```

Reply:
Ok sure, let's remove the todo though. We shouldn't keep todos which are no longer relevant.